compiler/rustc_parse/src/parser/diagnostics.rs

```diff
@@ -880,6 +880,97 @@ impl<'a> Parser<'a> {
         }
     }
 
+    pub(super) fn check_anonymous_enum(&mut self, span: Span, tys: &[P<Ty>]) -> PResult<'a, ()> {
+        use std::fmt::Write;
+        if tys.len() <= 1 {
+            return Ok(());
+        }
+
+        fn variant_name_and_ty<'a>(
+            ty: &'a Ty,
+            lifetimes: &mut FxHashSet<&'a str>,
+        ) -> Option<(String, String)> {
+            match &ty.kind {
+                TyKind::Path(_, path) => {
+                    let mut name = String::new();
+                    let mut ty_string = String::new();
+
+                    if let Some(seg) = path.segments.iter().last() {
+                        name.push_str(seg.ident.name.as_str());
```

Comment:
It feels like the only thing we need here is the `name` and for `ty_string` `format!("{ty}")` is enough, after folding the anon lifetimes to be named as `'lifetime`, in way similar to what is done here:

https://github.com/rust-lang/rust/blob/c3fce8e9379069daf50ac393918ab6e60b57f4ca/compiler/rustc_middle/src/ty/erase_regions.rs#L56-L69

Reply:
emmm, the `TyKind` in rustc_ast doesn't implement `Display`, if it's nesscery to implement `Display`, I will submit a new pull request to do that.

Reply:
I'm fairly certain there's a way to `pprint` `TyKind`, but it isn't through `Display` directly 🤔